src/ast/data.cr

```diff
@@ -6,13 +6,8 @@ module Mint
       def initialize(@input : String, @file : String)
       end
 
-      def to_s
-        "<#{file} #{@input[0, 10]}...>"
-      end
-
-      def to_s(io)
-        io << to_s
-        io
+      def to_s(io : IO)
+        io << '<' << file << ' ' << input[0, 10] << "...>"
```

Comment:
Curious why do we need to do this (`<<`) for every part?

Reply:
To avoid string interpolations.

Reply:
Sure! For in-depth explanation see [_Don't create intermediate strings when writing to an IO_](https://crystal-lang.org/reference/guides/performance.html#dont-create-intermediate-strings-when-writing-to-an-io) section in the crystal reference guide.